applications/SeismicApplication/custom_constitutive/uniaxial_menegotto_pinto.cpp

```diff
@@ -0,0 +1,242 @@
+//  KRATOS  ____       _               _
+//         / ___|  ___(_)___ _ __ ___ (_) ___
+//         \___ \ / _ \ / __| '_ ` _ \| |/ __|
+//          ___) |  __/ \__ \ | | | | | | (__
+//         |____/ \___|_|___/_| |_| |_|_|\___|
+//
+//  License:     BSD License
+//  license:     structural_mechanics_application/license.txt
+//
+//  Main authors: Mahmoud Zidan
+//
+
+// System includes
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "includes/properties.h"
+#include "custom_constitutive/uniaxial_menegotto_pinto.hpp"
+#include "seismic_application_variables.h"
+
+namespace Kratos {
+
+
+ConstitutiveLaw::Pointer UniaxialMenegottoPintoMaterialLaw::Clone() const
+{
+    return Kratos::make_shared<UniaxialMenegottoPintoMaterialLaw>(*this);
+}
+
+void UniaxialMenegottoPintoMaterialLaw::GetLawFeatures(Features& rFeatures)
+{
+    rFeatures.mStrainSize = 1;
+    rFeatures.mSpaceDimension = 3;
+}
+
+void UniaxialMenegottoPintoMaterialLaw::InitializeMaterialResponsePK2(Parameters& rValues)
+{
+    KRATOS_TRY
+    mTangentModulus = rValues.GetMaterialProperties()[STEEL_YOUNGS_MODULUS];
+    KRATOS_CATCH("")
+}
+
+void UniaxialMenegottoPintoMaterialLaw::CalculateMaterialResponsePK2(Parameters& rValues)
+{
+    KRATOS_TRY
+
+    Flags& r_options = rValues.GetOptions();
+    if (r_options.Is(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN)) {
+        Vector& r_strain_vector = rValues.GetStrainVector();
+
+        if (r_options.Is(ConstitutiveLaw::COMPUTE_STRESS)) {
+            Vector& r_stress_vector = rValues.GetStressVector();
+            CalculateStressResponsePK2(rValues, r_strain_vector, r_stress_vector);
+        }
+
+        if (r_options.Is(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR)) {
+            Matrix& r_constitutive_matrix = rValues.GetConstitutiveMatrix();
+            CalculateConstitutiveMatrix(rValues, r_constitutive_matrix);
+        }
+    }
+    else {
+        KRATOS_ERROR << "Only ELEMENT_PROVIDED_STRAIN is implemented" << std::endl;
+    }
+
+    KRATOS_CATCH("")
+}
+
+void UniaxialMenegottoPintoMaterialLaw::FinalizeMaterialResponsePK2(Parameters& rValues)
+{
+    KRATOS_TRY
+    mConvergedLoadingIndex  =                mLoadingIndex;
```

Comment:
This is not properly done, please look the plasticity laws implemented by @AlejandroCornejo. You don't save everything twice, you revaluate and update the converged value